services/submission/src/main/java/app/coronawarn/server/services/submission/validation/ValidSubmissionPayload.java

```diff
@@ -132,16 +136,45 @@ private boolean checkNoOverlapsInTimeWindow(List<TemporaryExposureKey> exposureK
 
       Integer[] sortedStartIntervalNumbers = exposureKeys.stream()
           .mapToInt(TemporaryExposureKey::getRollingStartIntervalNumber)
-          .sorted().boxed().toArray(Integer[]::new);
+          .distinct().sorted().boxed().toArray(Integer[]::new);
 
       for (int i = 1; i < sortedStartIntervalNumbers.length; i++) {
-        if ((sortedStartIntervalNumbers[i - 1] + EXPECTED_ROLLING_PERIOD) > sortedStartIntervalNumbers[i]) {
+        if ((sortedStartIntervalNumbers[i - 1] + maxRollingPeriod) > sortedStartIntervalNumbers[i]) {
           addViolation(validatorContext, String.format(
               "Subsequent intervals overlap. StartIntervalNumbers: %s", sortedStartIntervalNumbers));
           return false;
         }
       }
       return true;
     }
+
+    private boolean checkKeysCummulateEqualOrLessThanMaxRollingPeriodPerDay(List<TemporaryExposureKey> exposureKeys,
+        ConstraintValidatorContext validatorContext) {
+
+      HashMap<Integer, Integer> totalKeysPerDay = new HashMap<>();
+
+      for (TemporaryExposureKey exposureKey : exposureKeys) {
```

Comment:
Consider grouping by start interval number, summing up the rolling periods for each group and then checking if `anyMatch` for sum > 144. Should be a lot shorter and more memory efficient.